File: index/index_retrieve.js
import { TwelveLabs } from 'twelvelabs-js';

(async () => {
  const client = new TwelveLabs({ apiKey: process.env.API_KEY}); 

  const retrievedIndex = await client.index.retrieve('');
  console.log(`ID: ${retrievedIndex.id}`);
  console.log(`Name: ${retrievedIndex.name}`);
  console.log("Engines:");
  retrievedIndex.engines.forEach((engine, index) => {
    console.log(`  Engine ${index + 1}:`);
    console.log(`    Name: ${engine.name}`);
    console.log(`    Options: ${JSON.stringify(engine.options)}`);
  });
  console.log(`Video count: ${retrievedIndex.videoCount}`);
  console.log(`Total turation: ${retrievedIndex.totalDuration} seconds`);
  console.log(`Created at: ${retrievedIndex.createdAt}`);
  if (retrievedIndex.updatedAt) {
    console.log(`Updated at: ${retrievedIndex.updatedAt}`);
  }
})();

File: index/index-retrieve.py
from twelvelabs import TwelveLabs
import os

client = TwelveLabs(api_key=os.environ.get("API_KEY"))


retrieved_index = client.index.retrieve("")
print(f"ID: {retrieved_index.id}")
print(f"Name: {retrieved_index.name}")
print("Engines:")
for i, engine in enumerate(retrieved_index.engines, 1):
    print(f"  Engine {i}:")
    print(f"    Name: {engine.name}")
    print(f"    Options: {engine.options}")
print(f"Video count: {retrieved_index.video_count}")
print(f"Total duration: {retrieved_index.total_duration} seconds")
print(f"created at: {retrieved_index.created_at}")
if retrieved_index.updated_at:
    print(f"Updated at: {retrieved_index.updated_at}")
